### PR TITLE
unsafe re-order and make .ts > .js

### PR DESCRIPTION
**Pull request by bluelovers**
https://github.com/TypeStrong/ts-node/issues/648

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -399,9 +401,7 @@ export function register (opts: Options = {}): Register {
   const register: Register = { cwd, compile, getTypeInfo, extensions, cachedir, ts }
 
   // Register the extensions.
-  extensions.forEach(extension => {
-    registerExtension(extension, ignore, register, originalJsHandler)
-  })
+  registerHanlder(opts, extensions, ignore, register, originalJsHandler)
```
Comment (3mard) [inline src/index.ts]: Typo 

```suggestion
  registerHandler(opts, extensions, ignore, register, originalJsHandler)
```

File: src/index.ts [hunk 2]
```diff
@@ -415,6 +415,41 @@ function shouldIgnore (filename: string, ignore: RegExp[]) {
   return ignore.some(x => x.test(relname))
 }
 
+/**
+ * unsafe re-order and make .ts > .js
+ */
+function registerHanlder (
```
Comment (3mard) [inline src/index.ts]: Typo

```suggestion
function registerHandler (
```

**Comment by 3mard** [comment]
@blakeembrey Will this be merged ?

**Comment by blakeembrey** [comment]
@3mard Yes, if we refactor and add a test I'm ok merging this.

**Comment by 3mard** [comment]
@bluelovers will this be updated ?

**Comment by bluelovers** [comment]
@3mard

**Comment by 3mard** [comment]
@bluelovers Can you please add some unit tests ? ( ... incase the hack became absolute )

**Comment by bluelovers** [comment]
@3mard   i don't know how add a test for this

**Comment by blakeembrey** [comment]
This was landed in https://github.com/TypeStrong/ts-node/pull/837.

**Pull request closed without merging**